unit_testing/yaml.test.js

```diff
@@ -120,32 +122,41 @@ const checkFolder = (videoFormat, previousPath, folder) => describe(folder, () =
       }
     });
 
+    test('has valid YAML', () => {
+      yamlContents = contents.split("---")[1];
+      decodedYaml = yaml.safeLoad(yamlContents);
+    });
+
     // Uses PropTypes to validate the structure and types of all of the
     // parts of the YAML, including if there are keys that don't exist
     // in the definition
     test("has valid YAML layout and types", () => {
-      let yamlContents = contents.split("---")[1];
-      const frontYaml = yaml.safeLoad(yamlContents);
-      assertPropTypes(videoFormat, frontYaml, "YAML", fileName);
+      assertPropTypes(videoFormat, decodedYaml, "YAML", fileName);
     });
 
     test('title matches internal numbering', () => {
       // Get file name with leading zeros stripped
       let fileNumber = fileName.split('-')[0];
       fileNumber = fileNumber.replace(/^0+/, '');
 
-      let yamlContents = contents.split("---")[1];
-      const frontYaml = yaml.safeLoad(yamlContents);
-
       // Gets internal representation as a string
-      let videoString = frontYaml.video_number.toString();
+      let videoString = decodedYaml.video_number.toString();
       // If we're in a standalone
       if(videoString === fileNumber) return;
       // If the video has parts, check we match the last one only.
       let parts = fileNumber.split('.');
       if(videoString === parts[parts.length - 1]) return;
       throw new Error('Expected file numbering to match internal numbering');
-    })
+    });
+
+    test('has no contributions if it\'s in a multi-part video but not the first part', () => {
```

Comment:
If anyone can think of a better descriptor, that'd be great!

Reply:
Hmm, tough one... Maybe 'has valid contributions' but I can't think of anything better either.

Reply:
What about 'has no contributions for multi-part videos beyond part one'